src/Microsoft.Extensions.ML/PoolLoader.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.ObjectPool;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.ML;
+
+namespace Microsoft.Extensions.ML
+{
+    internal class PoolLoader<TData, TPrediction>: IDisposable
+                        where TData : class
+                        where TPrediction : class, new()
+    {
```

Comment:
Although this is an internal class, I'd still like to have some mild documentation. My intention is to treat devs like end users so that ML.NET can be more inclusive.

Reply:
Added a summary doc to this class.